### PR TITLE
Difficulty

### PR DESCRIPTION
**Pull request by adQuid**
**Brief Description of What This PR Does**

  Adds a simple hard mode, saved with the game so that a hard mode game will still be hard mode when loaded.

   I ran the translation update script, but I can't figure out how to make the text on the button match the strings in the locale file.

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [ ] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: locale/en.po [hunk 1]
```diff
@@ -1331,61 +1331,66 @@ msgstr "Art by {0}"
...
 msgid "Cancel"
 msgstr "Cancel"
 
 #: ../src/general/MainMenu.tscn:146
 msgid "NEW_GAME"
 msgstr "New Game"
 
-#: ../src/general/MainMenu.tscn:159 ../src/general/PauseMenu.tscn:71
+#: ../src/general/MainMenu.tscn:159
+#, fuzzy
```
Comment (hhyyrylainen) [inline locale/en.po:1347]: This fuzzy line marks this translation as in progress, so the game won't load it

**Comment by hhyyrylainen** [comment]
Why only just a hard mode? Why not also an easy mode, like asked for in this issue: https://github.com/Revolutionary-Games/Thrive/issues/575

@buckly90 might not appreciate the fact that this triples the balancing work... (especially when the microbe stage is not yet feature complete)

**Comment by adQuid** [comment]
The primary goal here was to get practice while I was blocked on other work by not knowing Godot. Since it does also offer something, I thought I might as well offer it to the team. Ideally there would be a game setup menu and such, with all the difficulty options you want (it's implemented as a simple float in this PR), but that would start taking time away from things I prioritize higher. I went with a hard mode instead of an easy mode for the other button out of personal taste, since I've personally found the game easy.

**Comment by buckly90** [comment]
I think if we rely on variable multipliers to game stats it would make balancing far less of a headache to keep balanced. (So if the base stats are changed then the hard/easy multipliers would reflect the changes accordingly). Hopefully that would mean we wouldn't have to devote too much attention to the alternative difficulties after initial fine tuning of the multipliers. It could also carry over well for custom difficulty settings should that be a thing.

Though as hh mentioned, I feel it may be best to tackle this once many of the big features are out of the way.

**Comment by hhyyrylainen** [comment]
I think that for difficulties, we should have json files that define a set of values for each difficulty level. That way whoever does the balancing can do it right.

But I think it is a bit early to add that. If this is to be added now this should store the game balance info (along with other new game options) in the `GameProperties` (and special care is needed to make sure the values are all loaded correctly, to not make old saves incompatible when new game balance variables are added). Also this kinda depends on getting the new game setup screen done: https://github.com/Revolutionary-Games/Thrive/issues/873

**Comment by adQuid** [comment]
It's hard to tell, but this is stored in the GameProperties: https://github.com/Revolutionary-Games/Thrive/pull/2330/files?file-filters%5B%5D=.cs#diff-f5a4b9c6963d01cf59494a8900f70a56273aba8bb950f1b1bc8dfd269368c036R18

**Comment by hhyyrylainen** [comment]
Ah, okay. The namespace schenanigans confused me. We don't use namespaces for general gameplay stuff.

**Comment by adQuid** [comment]
It seemed like the best way for stuff in the microbestage folder to be able to pull data from NewGameSettings, and the linter took control from there.

**Comment by hhyyrylainen** [comment]
Everything should be linked in just fine as long as the csproj properly references all files. Everything that is not in a namespace shares the global namespace.

**Comment by adQuid** [comment]
Ok, if you guys are interesting in seeing this PR go in I'll finagle with it, otherwise I'll drop it and find something else to work on. This is an easy PR to move forward from when it's obsolete, and an easy to just remake later once we have a setup screen.

**Comment by hhyyrylainen** [comment]
I would recommend working on something else for now, as difficulty options to me seem something that should be only added after all features and the "normal" difficulty is properly balanced.

**Pull request closed without merging**